Shopify/shopify-cli
# Perform node version checks with ToolVersionChecker module

### WHY are these changes introduced?

Fixes https://github.com/Shopify/script-service/issues/2202

### WHAT is this pull request doing?

Adds a new module which infra-layer classes can use to ensure the system tools have an appropriate version.

### How to test your changes?

create / push an assembly/typescript script project.
Tweak the various `MIN_{TOOL}_VERSION` in that language's project_creator and task_runner to observe the different behaviours.

### Update checklist
